diff -up iproute2-2.6.28/iproute2-2.6.28/Makefile.kernel iproute2-2.6.28/iproute2-2.6.28/Makefile
--- iproute2-2.6.28/iproute2-2.6.28/Makefile.kernel	2008-07-25 22:46:07.000000000 +0200
+++ iproute2-2.6.28/iproute2-2.6.28/Makefile	2008-08-12 14:19:58.000000000 +0200
@@ -3,6 +3,7 @@ SBINDIR=/sbin
 CONFDIR=/etc/iproute2
 DOCDIR=/share/doc/iproute2
 MANDIR=/share/man
+KERNEL_INCLUDE=/usr/include
 
 # Path to db_185.h include
 DBM_INCLUDE:=/usr/include
