--- iproute2/ip/iproute.c.initvar	2004-04-21 16:57:59.307562694 +0200
+++ iproute2/ip/iproute.c	2004-04-21 16:58:35.353103364 +0200
@@ -620,6 +620,9 @@
 		}
 		memset(rtnh, 0, sizeof(*rtnh));
 		rtnh->rtnh_len = sizeof(*rtnh);
+		rtnh->rtnh_ifindex = 0;
+		rtnh->rtnh_flags = 0;
+		rtnh->rtnh_hops = 0;
 		rta->rta_len += rtnh->rtnh_len;
 		parse_one_nh(rta, rtnh, &argc, &argv);
 		rtnh = RTNH_NEXT(rtnh);
