--- iproute2-2.6.33/ip/link_veth.c	2010-02-25 04:56:50.000000000 +0100
+++ iproute2-2.6.33/ip/link_veth.c.new	2010-10-12 17:25:01.159138225 +0200
@@ -27,7 +27,7 @@
 static int veth_parse_opt(struct link_util *lu, int argc, char **argv,
 			  struct nlmsghdr *hdr)
 {
-	char *name, *type, *link, *dev;
+	char *name = NULL, *type = NULL, *link = NULL, *dev = NULL;
 	int err, len;
 	struct rtattr * data;
 
