diff -up iproute-2.6.25/iproute2-2.6.25/Makefile.old iproute-2.6.25/iproute2-2.6.25/Makefile
--- iproute-2.6.25/iproute2-2.6.25/Makefile.old	2007-10-16 23:27:42.000000000 +0200
+++ iproute-2.6.25/iproute2-2.6.25/Makefile	2007-10-31 09:22:41.000000000 +0100
@@ -3,6 +3,7 @@ SBINDIR=/usr/sbin
 CONFDIR=/etc/iproute2
 DOCDIR=/usr/share/doc/iproute2
 MANDIR=/usr/share/man
+KERNEL_INCLUDE=/usr/include
 
 # Path to db_185.h include
 DBM_INCLUDE:=/usr/include
