diff -up iproute-2.6.23/iproute2-2.6.23/netem/Makefile.lib iproute-2.6.23/iproute2-2.6.23/netem/Makefile
--- iproute-2.6.23/iproute2-2.6.23/netem/Makefile.lib	2007-10-16 23:27:42.000000000 +0200
+++ iproute-2.6.23/iproute2-2.6.23/netem/Makefile	2007-10-31 09:31:07.000000000 +0100
@@ -19,9 +19,9 @@ stats: stats.c
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/usr/lib/tc
+	mkdir -p $(DESTDIR)/usr/share/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/usr/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)/usr/share/tc; \
 	done
 
 clean:
diff -up iproute-2.6.23/iproute2-2.6.23/tc/tc_util.c.lib iproute-2.6.23/iproute2-2.6.23/tc/tc_util.c
--- iproute-2.6.23/iproute2-2.6.23/tc/tc_util.c.lib	2007-10-16 23:27:42.000000000 +0200
+++ iproute-2.6.23/iproute2-2.6.23/tc/tc_util.c	2007-10-31 09:33:00.000000000 +0100
@@ -30,7 +30,7 @@ const char *get_tc_lib(void)
 
 	lib_dir = getenv("TC_LIB_DIR");
 	if (!lib_dir)
-		lib_dir = "/usr/lib/tc";
+		lib_dir = "/usr/share/tc";
 
 	return lib_dir;
 }
