diff -up iproute-2.6.26/iproute2-2.6.26/netem/Makefile.lib iproute-2.6.26/iproute2-2.6.26/netem/Makefile
--- iproute-2.6.26/iproute2-2.6.26/netem/Makefile.lib	2008-07-25 22:46:07.000000000 +0200
+++ iproute-2.6.26/iproute2-2.6.26/netem/Makefile	2008-08-12 14:31:54.000000000 +0200
@@ -20,9 +20,9 @@ stats: stats.c
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/lib/tc
+	mkdir -p $(DESTDIR)/share/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)/share/tc; \
 	done
 
 clean:
diff -up iproute-2.6.26/iproute2-2.6.26/tc/tc_util.c.lib iproute-2.6.26/iproute2-2.6.26/tc/tc_util.c
--- iproute-2.6.26/iproute2-2.6.26/tc/tc_util.c.lib	2008-07-25 22:46:07.000000000 +0200
+++ iproute-2.6.26/iproute2-2.6.26/tc/tc_util.c	2008-08-12 14:32:54.000000000 +0200
@@ -34,7 +34,7 @@ const char *get_tc_lib(void)
 
 	lib_dir = getenv("TC_LIB_DIR");
 	if (!lib_dir)
-		lib_dir = DESTDIR "/lib/tc";
+		lib_dir = DESTDIR "/share/tc";
 
 	return lib_dir;
 }
